src/app/core/header/header.component.spec.ts

```diff
@@ -5,8 +5,42 @@ import { HeaderModule } from './header.module';
 
 describe('HeaderComponent', () => {
   let shallow: Shallow<HeaderComponent>;
+  const filterClass = '.filter-labels';
 
   beforeEach(() => {
     shallow = new Shallow(HeaderComponent, HeaderModule);
   });
+
+  it('should emit downloadClicked when download button is clicked', async () => {
+    const { find, outputs } = await shallow.render();
+    find('.download').triggerEventHandler('click', {});
+    expect(outputs.downloadClicked.emit).toHaveBeenCalled();
+  });
+
+  it('should emit logoClicked when logo is clicked', async () => {
+    const { find, outputs } = await shallow.render();
+    find('.logo').triggerEventHandler('click', {});
+    expect(outputs.logoClicked.emit).toHaveBeenCalled();
+  });
+
+  it('should display the current sex', async () => {
+    const { find } = await shallow
+      .render({bind: {filters: { sex: 'Both', ageRange: [1, 110], BMIRange: [13, 83] }}});
```

Comment:
For the future only send the minimally required data. Keeps the logic simple and reduces complexity.